packages/cli/cloud/src/config/local.ts

```diff
@@ -0,0 +1,59 @@
+import path from 'path';
+import os from 'os';
+import fs from 'fs';
+import XDGAppPaths from 'xdg-app-paths';
+
+const APP_FOLDER_NAME = 'com.strapi.cli';
+
+export const CONFIG_FILENAME = 'config.json';
+
+export type LocalConfig = {
+  token?: string;
+};
+
+function checkDirectoryExists(directoryPath: string) {
+  try {
+    return fs.lstatSync(directoryPath).isDirectory();
+  } catch (e) {
+    return false;
+  }
+}
+
+// Determine storage path based on the operating system
+export function getTmpStoragePath() {
+  const storagePath = path.join(os.tmpdir(), APP_FOLDER_NAME);
+  if (!checkDirectoryExists(storagePath)) {
+    fs.mkdirSync(storagePath, { recursive: true });
+  }
+  return storagePath;
+}
+
+function getConfigPath() {
+  const configDirs = XDGAppPaths(APP_FOLDER_NAME).configDirs();
+  const configPath = configDirs.find(checkDirectoryExists);
+
+  if (!configPath) {
+    fs.mkdirSync(configDirs[0], { recursive: true });
+    return configDirs[0];
+  }
+  return configPath;
+}
+
+export function getLocalConfig(): LocalConfig {
+  const configPath = getConfigPath();
+  const configFilePath = path.join(configPath, CONFIG_FILENAME);
+  if (!fs.existsSync(configFilePath)) {
+    return {};
+  }
+  try {
+    return JSON.parse(fs.readFileSync(configFilePath, 'utf8'));
+  } catch (e) {
+    return {};
+  }
+}
+
+export function saveLocalConfig(data: LocalConfig) {
```

Comment:
I think it's going to be difficult to use fs in async mode, as we don't have top-level await and I've just needed to create and store a deviceId in the configuration file for use during event tracking.

Reply:
You can make all the functions async and use a then/catch at the root level anyway.

Reply:
It would make everything async including `buildStrapiCloudCommands`, then it would be used in the strapi/core/commands CLI builder and I'm unsure of how commander would react with something loaded async in it? 🤔

Reply:
Hum you have a, implementation issue then,

`buildStrapiCloudCommands` is calling the `initCloudCLIConfig` fn (which is async) without being async itself or doing a then/catch. You can get unhandled promise rejections right now. 

I don't know what you would like to do if it fails but depending either make it sync or just make the buildStrapiCloudCommands async, I don't see a pbl doing that, as long as you call parse() after this has resolved